### DISCUSSION TITLE
Sunny Portal empfängt seit gestern keine Daten mehr?

### DISCUSSION THREAD

**Discussion by LurchiStromberg** [camueller/SmartApplianceEnabler#426] [Q&A]
Moin!

Kann es sein, dass man im Sunny Portal keine Daten mehr für die Verbraucher sieht in der Tagesübersicht?
Seit gestern ist bei mir überall eine "0"

Bei "aktuell" werden aber die aktuellen Werte der Verbraucher angezeigt.

Wieder eine SMA Glanzleistung?

EDIT:
Ist das wirklich nur bei mir so?

Alle Verbraucher werden in der Geräteübersicht im Portal unter "Einstellungen" als aktiv mit grün angezeigt.
Auch jetzt steht der Tagesverbrauch bei sämtlichen Verbrauchern auf 0 Wh.

Letze komplette Daten hatte ich am Mittwoch.... Dana nicht mehr.
Auch SAE läuft einwandfrei, bekomme zumindest keine Fehlermeldung.

Woran könnte das liegen, dass das Sunny Portal zwar aktuelle Werte anzeigt, diese aber nicht mehr abspeichert?

**Comment by camueller** [comment]
Bei mir werden die Tageswerte für gestern (Du hattest ja gestern abend gepostet) und auch heute anggezeigt.

Aber Deine Analyse ist korrekt: Wenn die Werte unter "Aktuell" im Portal angezeigt werden, aber nicht die Tagessummen, kann es eigentlich nur am SHM/Portal liegen.

Verwendest Du einen SHM 1.0 oder 2.0?

**Reply by LurchiStromberg** [reply]
> Verwendest Du einen SHM 1.0 oder 2.0?

Hallo!
SHM 2.0

Hier mal Screenshots. Heute wieder das Gleiche.
Die aktuellen Daten werden aber tatsächlich erst nachdem ich auf F5 im Browser drücke angezeigt, beim ersten Mal nach den Einloggen ist die Liste leer.

<a href="https://ibb.co/8z29K3k"><img src="https://i.ibb.co/xmG6CvZ/sunny3.jpg" alt="sunny3" border="0"></a>
<a href="https://ibb.co/NFGzrXp"><img src="https://i.ibb.co/3TPHvwW/Sunny2-aktuell.jpg" alt="Sunny2-aktuell" border="0"></a>
<a href="https://ibb.co/Fm8fR9h"><img src="https://i.ibb.co/fqrKjbX/Sunny1.jpg" alt="Sunny1" border="0"></a>

Bin da im Moment etwas ratlos, hatte allerdings erst vor kurzem (war das am Mittwoch?) SAE 2.1.0 installiert.
Nur wenn im Portal aktueller Werte angezeigt werden, müsste es doch eigentlich gehen.....
Warum das Portal die Werte nicht "speichert" ist mir ein Rätsel.
Evtl. gehe ich noch einmal auf eine ältere Version von SAE zurück, hatte davor glaube ich die 1.7.0, seit 2 Jahren, ohne Probleme am Laufen.

**Reply by LurchiStromberg** [reply]
Hatte vorher die 1.5.1
"Blöd", dass man die Appliances.xml nicht zwischen den Versionen kopieren kann, die hat sich ja verändert.
Mit der <Appliances xmlns="http://github.com/camueller/SmartApplianceEnabler/v2.0"> erkennt ja der 1.5.1 nicht die Geräte...

**Reply by camueller** [reply]
Ich denke nicht, dass das eingangs von Dir beschriebene Problem durch ein SAE-Downgrade behoben werden kann.

Du kannst zwischen den Versionen wechseln, aber Du musst jeder Version eine `Appliances.xml` geben, die zu ihr passt. Wenn Du also wieder auf die 1.5.1 zurück gehst, musst Du die (hoffentlich vor dem Update gesicherte) `Appliances.xml` wieder zurückspielen, die Du bis zum Update verwendet hast.

**Reply by LurchiStromberg** [reply]
Habe jetzt erst gesehen, dass die 2.1.0 MQTT basiert ist. Soweit kein Problem, ich nutze nur keinen RASPI.
Habe einen Ubuntu Server, da laufen auch noch andere Sachen drauf, u. a. auch mit MQTT.
Evtl. funkt da etwas dazwischen...
Wobei, wenn doch aktuelle Daten bei SMA ankommen?

Habe jetzt mal die 1.6.21 installiert....
Durch die Aktion mit meiner Wärmepumpe sind natürlich die Alliances.xml nicht mehr brauchbar, da andere Seriennummern etc.
Naja, habe jetzt erst einmal 1 Device angelegt und gucke, ob dieses nun im Sunny Portal "gezählt" wird....

Ist schon was dran an dem Spruch "never change a running System" ;)

**Reply by LurchiStromberg** [reply]
Da scheint aber bei SMA irgendwas quer zu sein.
Habe nun nur den sOnOff vom Wasserbett in SAE eingerichtet, als einziges Gerät.
Im Sunny Portal werden trotzdem weiterhin alle anderen online angezeigt.....

Muss ich wohl mal SMA anfunken....

**Reply by LurchiStromberg** [reply]
Seit 18 Uhr, wie durch ein Wunder, funktioniert die Verbrauchsbilanz wieder und das mit Version 2.1.0.

Könnte das evtl. am Abfrage Intervall liegen? Den habe ich auf 10 Sekunden. Meine nur, dass es schon seit Jahren auf 10 Sekunden gestanden hat.
Musste beim Wechsel von er alten auf die neue Version alle Geräte neu anlegen.

Streikt das Sunny Portal bei 10 Sekunden?

**Comment by Doenerteller** [comment]
Das Abfrageintervall besagt nur wie oft der SAE beim Stromzähler die Werte anfragt. Es besagt nicht, wie oft die Daten an SMA gemeldet werden. Ich glaube der Home Manager holt sich nur alle Minute die Daten beim SAE ab.

Was ich jetzt allerdings nicht weiß ist, ob wenn du ein Abfrageinterval von z.B 10 Sekunden hast, ob der Home Manger nach einer Minute dann 6 Werte im 10 Sekunden Abstand bekommt, oder ob der NUR DEN LETZEN Wert von kurz vor seiner Kontaktaufnahme bekommt.

Das müsstest du/man ja leicht rausfinden können: geht mal mit dem Browser auf die SEMP Adresse des SAE. Wieviele Werte bekommst du bei einem Zähler, bei dem das Abfrageinterval kleiner als 60 Sekunden ist, angezeigt? Einen oder mehrere?
Denn nur was der Home Manger hier sieht kann er SMA weitergeben.

Grundsätzlich habe ich persönlich nicht das größte Vertrauen in SMA und ihr Portal. Das ist so oft auf die ein oder andere Art und Weise kaputt. Die Verbrauchsdaten werden teilweise auch erst Stunden später dort sichtbar. Meistens wartet man einfach einszwei Tage und dann ist es wieder in Ordnung.

**Comment by camueller** [comment]
> Was ich jetzt allerdings nicht weiß ist, ob wenn du ein Abfrageinterval von z.B 10 Sekunden hast, ob der Home Manger nach einer Minute dann 6 Werte im 10 Sekunden Abstand bekommt, oder ob der NUR DEN LETZEN Wert von kurz vor seiner Kontaktaufnahme bekommt.

Der SHM bekommt nur einen Wert. Wenn der SAE vom Zähler die Leistung abfragt, ist es der Durchschnittswert aller Abfragen der letzten Minute. Wenn der SAE vom Zähler den Zählerstand abfragt (alle 60s, kann nicht anders konfiguriert werden), wird die Leistung aus der Differenz seit der letzten Zählerstandsabfrage berechnet.

**Comment by Doenerteller** [comment]
Saugut! Danke!

**Comment by Doenerteller** [comment]
Vielelicht weiß es jemand hier (bin noch nicht fündig geworden bei Shelly, die Suche geht weiter...): ist das bei einem Shelly Pro 1PM der Zählerstand?

![grafik](https://user-images.githubusercontent.com/119921647/212852768-b9b7c94f-d9a1-4fe7-b115-34fc177a63be.png)

Welche Einheit ist das? Wattminute? 

Und wäre dashier der korrekte Pfad für "Extraktion"?

$.switch:0.aenergy.total

Danke!

**Comment by Doenerteller** [comment]
Ich denke ich hab's gefunden. Die Shelly Doku verwirrt mich hals Laien mehr, als dass sie mir hilft, aber ich glaube hier bin ich richtig

![grafik](https://user-images.githubusercontent.com/119921647/212858097-7d221aab-567c-4e59-8a54-901fcb7d04c8.png)


Wäre nett, wenn mir noch jemand bezüglich des Pfades behilflich sein könnte. Hier rate ich auch nur... Den Umrechnungsfaktor werde ich hin wohl bekommen, das ist ja nur normale Mathematik.

Ich könnte es natürlilch auch einfach ausprobieren...

**Comment by camueller** [comment]
> Wäre nett, wenn mir noch jemand bezüglich des Pfades behilflich sein könnte. Hier rate ich auch nur...

Siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/ValueExtraction_DE.md

**Comment by Doenerteller** [comment]
Danke. Dann lag ich ja richtig.
